getting-started/bookshelf/pubsub_worker/worker.go

```diff
@@ -29,27 +28,29 @@ var (
 	countMu sync.Mutex
 	count   int
 
-	booksClient *books.Service
-	pubSubCtx   context.Context
+	booksClient  *books.Service
+	subscription *pubsub.SubscriptionHandle
 )
 
 func main() {
-	var err error
+	ctx := context.Background()
 
-	pubSubCtx, err = bookshelf.PubSubCtx()
-	if err != nil {
-		log.Fatal(err)
+	if bookshelf.PubsubClient == nil {
+		log.Fatal("You must configure the Pub/Sub client in config.go before running pubsub_worker.")
 	}
 
+	var err error
 	booksClient, err = books.New(http.DefaultClient)
 	if err != nil {
 		log.Fatalf("could not access Google Books API: %v", err)
 	}
 
 	// ignore returned errors, which will be "already exists". If they're fatal
 	// errors, then following calls (e.g. in the subscribe function) will also fail.
-	pubsub.CreateTopic(pubSubCtx, bookshelf.PubSubTopic)
-	pubsub.CreateSub(pubSubCtx, subName, bookshelf.PubSubTopic, 0, "")
+	bookshelf.PubsubClient.NewTopic(ctx, bookshelf.PubsubTopicID)
+	bookshelf.PubsubClient.NewSubscription(ctx, subName, bookshelf.PubsubClient.Topic(bookshelf.PubsubTopicID), 0, nil)
```

Comment:
I think it would be cleaner to pass in the TopicHandle which is returned from NewTopic here.

Of course, that would mean that you should also check the error returned from NewTopic, but I think that would be good practice anyway.